src/main/java/com/minecolonies/entity/ai/citizen/farmer/EntityAIWorkFarmer.java

```diff
@@ -21,520 +21,550 @@
-        return true;
-    }
-
-    /**
-     * Checks if the crop should be harvested.
-     *
-     * @param position the position to check.
-     * @return true if should be hoed.
-     */
-    private boolean shouldHarvest(@NotNull final BlockPos position)
-    {
-        final IBlockState state = world.getBlockState(position.up());
-
-        if (state.getBlock() instanceof IGrowable && state.getBlock() instanceof BlockCrops)
-        {
-            @NotNull final BlockCrops block = (BlockCrops) state.getBlock();
-            return !block.canGrow(world, position.up(), state, false);
-        }
-
-        return false;
-    }
-
-    /**
-     * This method allows us to harvest crops and leave the plant there.
-     * Credit goes to RightClickHarvest mod
-     */
-    private boolean harvestCrop(final BlockPos position)
-    {
-        final IBlockState curBlockState = world.getBlockState(position);
-
-        if (!(curBlockState.getBlock() instanceof IGrowable) || !(curBlockState.getBlock() instanceof BlockCrops))
-        {
-            return false;
-        }
-
-        final BlockCrops crops = (BlockCrops) curBlockState.getBlock();
-
-        if (!crops.isMaxAge(curBlockState))
-        {
-            return false;
-        }
-
-        final ItemStack tool = worker.getHeldItemMainhand();
-
-        //calculate fortune enchantment
-        final int fortune = Utils.getFortuneOf(tool);
-
-        final List<ItemStack> drops = crops.getDrops(world, position, curBlockState, fortune);
-
-        world.setBlockState(position, crops.withAge(0));
-
-        //add the drops to the citizen
-        for (final ItemStack item : drops)
-        {
-            InventoryUtils.setStack(worker.getInventoryCitizen(), item);
-        }
-
-        this.incrementActionsDone();
-        return true;
-    }
-
-    /**
-     * Checks if the ground should be hoed and the block above removed.
-     *
-     * @param position the position to check.
-     * @param field    the field close to this position.
-     * @return true if should be hoed.
-     */
-    private boolean shouldHoe(@NotNull final BlockPos position, @NotNull final Field field)
-    {
-        return !field.isNoPartOfField(world, position)
+ {
+     /**
+      * The standard delay the farmer should have.
+      */
+     private static final int     STANDARD_DELAY      = 5;
+     /**
+      * The bonus the farmer gains each update is level/divider.
+      */
+     private static final int     DELAY_DIVIDER       = 10;
+     /**
+      * Changed after finished harvesting in order to dump the inventory.
+      */
+     private              boolean shouldDumpInventory = false;
+     /**
+      * The offset to work at relative to the scarecrow.
+      */
+     @Nullable
+     private BlockPos workingOffset;
+     /**
+      * The delay the farmer should have each action: hoeing, planting, harvesting.
+      */
+     private int workingDelay = STANDARD_DELAY - this.worker.getLevel() / DELAY_DIVIDER;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Private field 'workingDelay' could be made final; it is only initialized in the declaration or constructor. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3AImmutableField)